src/session-pool.ts

```diff
@@ -262,6 +262,36 @@ export function isCreateSessionPermissionError(
   );
 }
 
+/**
+ * Checks whether the given error is a 'Could not load the default credentials' error.
+ * @param {Error} error The error to check.
+ * @return {boolean} True if the error is a 'Could not load the default credentials' error, and otherwise false.
+ */
+export function isDefaultCredentialsNotSetError(
+  error: grpc.ServiceError | undefined
+): boolean {
+  return (
+    error !== undefined &&
+    error.message.includes('Could not load the default credentials')
+  );
+}
+
+/**
+ * Checks whether the given error is a 'Project Id not set in Environment' error.
```

Comment:
nit: could we make the error description in the comment correspond with the one that is actually being tested in the method?

Reply:
`Project Id not set in Environment` is actual message getting while Project Id is not set in environment.

Reply:
Then the check below is wrong, because that one reads:

```typescript
    error.message.includes(
      'Unable to detect a Project Id in the current environment'
    )
```

Reply:
@olavloite  my bad below is the actual message getting while project Id or credential not set:
1.  `Unable to detect a Project Id in the current environment`
2.  `Could not load the default credentials`

I will make changes in method description accordingly.

Reply:
Done!!